packages/devextreme-cli/src/templates/react/application/src/layouts/side-nav-inner-toolbar/side-nav-inner-toolbar.js

```diff
@@ -2,180 +2,123 @@ import Button from 'devextreme-react/button';
+        : MenuStatus.Closed
     );
-  }
-
-  componentDidMount() {
-    subscribe(this.updateDrawer);
-  }
-
-  componentWillUnmount() {
-    unsubscribe(this.updateDrawer);
-  }
-
-  closeDrawer = () => {
-    if (!this.state.shaderEnabled) {
-      return false;
-    }
-
-    this.setState({ menuOpened: false });
-    return true;
-  }
-
-  updateDrawer = () => {
-    this.setState({ ...this.drawerConfig });
-  };
-
-  toggleMenu = ({ event }) => {
-    this.setState(({ menuOpened }) => {
-      return { menuOpened: !menuOpened };
-    });
     event.stopPropagation();
-  };
-
-  get scrollView() {
-    return this.scrollViewRef.current.instance;
-  }
-
-  get drawerConfig() {
-    const isXSmall = sizes()['screen-x-small'];
-    const isLarge = sizes()['screen-large'];
+  }, []);
 
-    return {
-      menuMode: isLarge ? 'shrink' : 'overlap',
-      menuRevealMode: isXSmall ? 'slide' : 'expand',
-      minMenuSize: isXSmall ? 0 : 60,
-      shaderEnabled: !isLarge
-    };
-  }
+  const temporaryOpenMenu = useCallback(() => {
+    setMenuStatus(
+      prevMenuStatus => prevMenuStatus === MenuStatus.Closed
+        ? MenuStatus.TemporaryOpened
+        : prevMenuStatus
+    );
+  }, []);
 
-  get hideMenuAfterNavigation() {
-    const { menuMode, temporaryMenuOpened } = this.state;
-    return menuMode === 'overlap' || temporaryMenuOpened;
-  }
+  const onOutsideClick = useCallback(() => {
+    setMenuStatus(
+      prevMenuStatus => prevMenuStatus !== MenuStatus.Closed && !isLarge
+        ? MenuStatus.Closed
+        : prevMenuStatus
+    );
+  }, [isLarge]);
 
-  navigationChanged = event => {
-    const path = event.itemData.path;
-    const pointerEvent = event.event;
+  const onNavigationChanged = useCallback(({ itemData: { path }, event, node }) => {
+    if (menuStatus === MenuStatus.Closed || !path || node.selected) {
+      event.preventDefault();
+      return;
+    }
 
-    if (path && this.state.menuOpened) {
-      if (event.node.selected) {
-        pointerEvent.preventDefault();
-      } else {
-        this.props.history.push(path);
-        this.scrollView.scrollTo(0);
-      }
+    history.push(path);
+    scrollViewRef.current.instance.scrollTo(0);
 
-      if (this.hideMenuAfterNavigation) {
-        this.setState({
-          menuOpened: false,
-          temporaryMenuOpened: false
-        });
-        pointerEvent.stopPropagation();
-      }
-    } else {
-      pointerEvent.preventDefault();
+    if (!isLarge || menuStatus === MenuStatus.TemporaryOpened) {
+      setMenuStatus(MenuStatus.Closed);
+      event.stopPropagation();
     }
-  };
-
-  navigationClick = () => {
-    this.setState(({ menuOpened }) => {
-      return !menuOpened
-        ? {
-          temporaryMenuOpened: true,
-          menuOpened: true
-        }
-        : {};
-    });
-  };
+  }, [history, menuStatus, isLarge]);
+
+  return (
+    <div className={'side-nav-inner-toolbar'}>
+      <Drawer
+        className={'drawer' + patchCssClass}
```

Comment:
```suggestion
        className={['drawer', patchCssClass].join(' ')}
```
And fix the `patchCssClass` (remove leading space)

Reply:
And the outer layout as well (-: